unitOneAssessmentPractice.js

```diff
@@ -81,47 +143,75 @@ let assert = require('assert')
 // Write a function called doubleAllElements that doubles each number in an array
 // Your function should use a higher-ordered function (e.g map, filter, reduce, every, sort) in its implementation
 // Your output should be in the same order as the input
+  const doubleAllElements = (arr) => {
+    return arr.map((el) => el * 2)
+    
+  }
 
 // Uncomment out the next line to test your solution
-// runQ10Tests()
+runQ10Tests()
 
 
 // Question Eleven:
 
 // Write a function called onlyLongStrings that removes all strings with 3 or fewer characters
 // Your function should use a higher-ordered function (e.g map, filter, reduce, every, sort) in its implementation
 // Your output should be in the same order as the input
-
+  const onlyLongStrings = (arr) => {
+    return arr.filter((el) => el.length > 3)
+  }
 // Uncomment out the next line to test your solution
-// runQ11Tests()
+runQ11Tests()
 
 // Question Twelve:
 
 // Write a function called containsOnlyNumbers that returns whether or not an array contains only numbers
 // Your function should use a higher-ordered function (e.g map, filter, reduce, every, sort) in its implementation
 // Hint: the isNaN() function will tell you whether something is not a number
-
+  const containsOnlyNumbers = (arr) => {
+    return arr.every((el) => !isNaN(el))
+  }
 // Uncomment out the next line to test your solution
-// runQ12Tests()
+runQ12Tests()
 
 // Question Thirteen:
 
 // Make a class called Person that has two properties set by the constructor named age and name
 // Give it a method called isALegalAdult which returns true if the age is at least 18
+  class Person {
+    constructor(age,name){
+      this.age = age
+      this.name = name
+    }
+    isALegalAdult() {
+     return this.age >= 18 ? true : false
+    }
+  }
 
-// runQ13Tests()
+runQ13Tests()
 
 // Question Fourteen:
 
 // Write a function called getAllAdults that takes in an array of Person objects and returns an array with only Person objects with an age of at least 18
-
-// runQ14Tests()
+  const getAllAdults = (arr) => {
+    return arr.filter((el) => {
+      return el["age"] >= 18
+    })
+  }
+runQ14Tests()
 
 // Question Fifteen:
 
 // Write a function called getAllNames that takes in an array of Person objects and returns a string with all of the names joined together with a ","
+  const getAllNames = (arr) => {
+    let str =""
+    arr.forEach((el) => {
+      str += (el.name + ",")
```

Comment:
The reason the tests for this question are failing is the comma after the last element. We want to have a comma after each element except last. What you could do is instead of creating a string you could create a new array and push all the el.name in it and then join it with .join(",")